cpp/src/arrow/csv/parser.cc

```diff
@@ -158,7 +173,7 @@ class ResizableValueDescWriter : public ValueDescWriter<ResizableValueDescWriter
   void PushValue(ParsedValueDesc v) {
     if (ARROW_PREDICT_FALSE(values_size_ == values_capacity_)) {
       values_capacity_ = values_capacity_ * 2;
-      ARROW_CHECK_OK(values_buffer_->Resize(values_capacity_ * sizeof(*values_)));
+      status_ &= values_buffer_->Resize(values_capacity_ * sizeof(*values_));
       values_ = reinterpret_cast<ParsedValueDesc*>(values_buffer_->mutable_data());
     }
     values_[values_size_++] = v;
```

Comment:
If the `Resize()` failed, is `values_[values_size_++] = v` safe?

Reply:
Hmm, that's a good point. We might adopt the same kind of approach as in the presized writer below.